src/main/cpp/vision/DragonQuest.cpp

```diff
@@ -96,45 +105,55 @@ void DragonQuest::ZeroPosition()
 
 void DragonQuest::DataLog(uint64_t timestamp)
 {
-    Log3DPoseData(timestamp, DragonDataLoggerSignals::PoseSingals::CURRENT_CHASSIS_QUEST_POSE3D, GetEstimatedPose());
+    Log2DPoseData(timestamp, DragonDataLoggerSignals::PoseSingals::CURRENT_CHASSIS_QUEST_POSE2D, GetEstimatedPose());
+    auto field = DragonField::GetInstance();
+    field->AddPose("Quest", GetEstimatedPose());
 }
 
 void DragonQuest::RefreshNT()
 {
     m_posTopic = m_networktable.get()->GetDoubleArrayTopic("position");
     m_rotationTopic = m_networktable.get()->GetDoubleArrayTopic("eulerAngles");
+    m_frameCountTopic = m_networktable.get()->GetIntegerTopic("frameCount");
+
+    SetIsConnected();
 }
 
 void DragonQuest::SetRobotPose(const frc::Pose2d &pose)
 {
-    frc::Pose3d p3d{frc::Pose3d(pose)};
-    m_xOffset += p3d.X().to<double>();
-    m_yOffset += p3d.Y().to<double>();
-    m_zOffset += p3d.Z().to<double>();
-    units::degree_t roll = p3d.Rotation().X();
-    units::degree_t pitch = p3d.Rotation().Y();
-    units::degree_t yaw = p3d.Rotation().Z();
-
-    m_rollOffset += roll.to<double>();
-    m_pitchOffset += pitch.to<double>();
-    m_yawOffset += yaw.to<double>();
-    m_hasreset = true;
+    if (!m_hasreset)
+    {
+        auto x = (pose.X() + m_mountingXOffset);
+        auto y = (pose.Y() + m_mountingYOffset);
+        auto rot = (pose.Rotation().Degrees() + m_mountingYaw);
+
+        m_initialPosePublisher.Set(std::array<double, 3>{x.value(), y.value(), rot.value()});
+
+        m_questTransform = pose - frc::Pose2d{x, y, frc::Rotation2d(rot)};
+
+        if (m_questMiso.Get() != 99)
+        {
+            sleep(1);
+            m_questMosi.Set(2);
+        }
+        m_hasreset = true;
+    }
 }
 
 DragonVisionPoseEstimatorStruct DragonQuest::GetPoseEstimate()
 {
     DragonVisionPoseEstimatorStruct str;
-    // if (!m_hasreset || !IsConnected())
-    // {
-    //     str.m_confidenceLevel = DragonVisionPoseEstimatorStruct::ConfidenceLevel::NONE;
-    // }
-    // else
-    // {
-    //     str.m_confidenceLevel = DragonVisionPoseEstimatorStruct::ConfidenceLevel::NONE;
-    //     str.m_visionPose = GetEstimatedPose().ToPose2d();
-    //     str.m_stds = wpi::array{m_stdxy, m_stdxy, m_stddeg};
-    //     str.m_timeStamp = units::time::second_t(GetTimeStamp());
-    // }
-    str.m_confidenceLevel = DragonVisionPoseEstimatorStruct::ConfidenceLevel::NONE;
+    Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("DragonQuest"), string("IsConnected"), m_isConnected);
+
+    if (!m_hasreset || !m_isConnected)
+    {
+        str.m_confidenceLevel = DragonVisionPoseEstimatorStruct::ConfidenceLevel::NONE;
+    }
+    else
+    {
+        str.m_confidenceLevel = DragonVisionPoseEstimatorStruct::ConfidenceLevel::NONE;
```

Comment:
I assume at some point we will set the confidence level here to something other than none.  Correct?

Reply:
That is correct, we want to look at the data in hte background and make sure it is good. Then turn up the confidence once we  believe in it